rust/main/chains/hyperlane-sealevel/src/merkle_tree_hook.rs

```diff
@@ -2,26 +2,63 @@ use std::ops::RangeInclusive;
 
 use async_trait::async_trait;
 use derive_new::new;
+use tracing::instrument;
+
 use hyperlane_core::{
-    accumulator::incremental::IncrementalMerkle, ChainCommunicationError, ChainResult, Checkpoint,
-    HyperlaneChain, HyperlaneMessage, Indexed, Indexer, LogMeta, MerkleTreeHook,
+    ChainCommunicationError, ChainResult, Checkpoint, CheckpointAtBlock, HyperlaneChain,
+    HyperlaneMessage, IncrementalMerkleAtBlock, Indexed, Indexer, LogMeta, MerkleTreeHook,
     MerkleTreeInsertion, ReorgPeriod, SequenceAwareIndexer,
 };
 use hyperlane_sealevel_mailbox::accounts::OutboxAccount;
-use tracing::instrument;
 
 use crate::{SealevelMailbox, SealevelMailboxIndexer};
 
 #[async_trait]
 impl MerkleTreeHook for SealevelMailbox {
     #[instrument(err, ret, skip(self))]
     #[allow(clippy::blocks_in_conditions)] // TODO: `rustc` 1.80.1 clippy issue
-    async fn tree(&self, reorg_period: &ReorgPeriod) -> ChainResult<IncrementalMerkle> {
+    async fn tree(&self, reorg_period: &ReorgPeriod) -> ChainResult<IncrementalMerkleAtBlock> {
+        assert!(
+            reorg_period.is_none(),
+            "Sealevel does not support querying point-in-time"
+        );
+
+        self.get_tree().await
+    }
+
+    #[instrument(err, ret, skip(self))]
+    #[allow(clippy::blocks_in_conditions)] // TODO: `rustc` 1.80.1 clippy issue
+    async fn latest_checkpoint(
+        &self,
+        reorg_period: &ReorgPeriod,
+    ) -> ChainResult<CheckpointAtBlock> {
         assert!(
             reorg_period.is_none(),
             "Sealevel does not support querying point-in-time"
         );
 
+        self.get_latest_checkpoint().await
+    }
+
+    #[instrument(err, ret, skip(self))]
+    #[allow(clippy::blocks_in_conditions)] // TODO: `rustc` 1.80.1 clippy issue
+    async fn latest_checkpoint_at_block(&self, _height: u64) -> ChainResult<CheckpointAtBlock> {
```

Comment:
add doc comment to indicate that the height doesn't influecne the result due to svm limitations

Reply:
Added